Add dg scaffold build-artifacts

benpankow opened the pull request:
## Summary

Adds a dg command to scaffold a Dockerfile for building a dg project.

`dg scaffold github-actions` now relies on having a Dockerfile in the serverless context.

## Test Plan

New unit test, update hybrid actions tests.

benpankow commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29335?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#29335** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29335?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29335?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#29065** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29065?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#28962** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/28962?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

gibsondan reviewed (changes requested):
one thing inline otherwise looks good

We should decide if we want dg deploy to error if this is not set or scaffold it for you if it is not there like it does currently (historically serverless didn't even let you specify your own dockerfile which makes things a bit tricky)

benpankow commented on python_modules/libraries/dagster-dg/dagster_dg/cli/scaffold.py:
```diff
@@ -193,6 +193,52 @@ def scaffold_workspace_command(
+        return Path(check.not_none(dg_context.build_config["directory"])) / "Dockerfile"
+    else:
+        return Path.cwd() / "Dockerfile"
+
```
updated

gibsondan reviewed (changes requested):
wait sorry it looks like this changes the behavior in deploy_session to no longer use the merged build config that potentially incorporates the build config the workspace - that logic should move into get_dockerfile_path

benpankow commented:
> wait sorry it looks like this changes the behavior in deploy_session to no longer use the merged build config that potentially incorporates the build config the workspace - that logic should move into get_dockerfile_path

good catch, updated

> less blocking comment - there was some discussion about potentially having a single command that scaffolds both this and the build.yaml (and the container_context.yaml once we add that?) - so all your build artifacts are created at once?

updated to scaffold build.yaml as well

benpankow commented:
Sorry for all the back-and-forth here. Updated to scaffold everything at the project level for now, also added some tests combining with `dg deploy` in the workspace and project contexts.

gibsondan reviewed (approved):
sweet, lets put this through its paces this afternoon?
